app/src/editor/canvas/components/ModuleSearch.jsx

```diff
@@ -120,9 +177,91 @@ export class ModuleSearch extends React.PureComponent<Props, State> {
         }
     }
 
+    getMappedModuleTree = (search: string) => {
+        const { allModules } = this.state
+        const modules = moduleTreeSearch(allModules, search)
+            .sort((a, b) => (a.path ? a.path.localeCompare(b.path) : 0))
+
+        const mapKeys = Object.keys(categoryMapping)
+
+        modules.forEach((m) => {
+            if (mapKeys.includes(m.path)) {
+                const newPath = categoryMapping[m.path]
+                m.path = newPath
+                m.pathArray = this.mapPathStringToArray(newPath)
+            }
+        })
+
+        return modules
+    }
+
+    mapPathStringToArray = (pathString: string) => (
+        pathString.split(', ')
+    )
+
+    mapPathArrayToString = (pathArray: Array<string>) => (
+        pathArray.join(': ')
+    )
+
+    renderMenu = () => {
+        const modules = this.getMappedModuleTree('')
+
+        // Form category tree
+        const categoryTree: { [string]: CategoryType } = {}
+        modules.forEach((m) => {
+            if (categoryTree[m.path] == null) {
+                categoryTree[m.path] = {
+                    name: m.path,
+                    modules: [m],
+                }
+            } else {
+                categoryTree[m.path].modules.push(m)
+            }
+        })
+        // https://github.com/facebook/flow/issues/2221
+        // $FlowFixMe Object.values() returns mixed[]
+        const categories: Array<CategoryType> = Object.values(categoryTree)
+
+        // $FlowFixMe "Missing type annotation for U"
+        return categories.map((category) => (
+            <React.Fragment key={category.name}>
+                <ModuleMenuCategory category={category} onItemSelect={this.onSelect} />
+            </React.Fragment>
+        ))
+    }
+
+    renderSearchResults = () => {
+        const { matchingModules, matchingStreams } = this.state
+        return (
+            <React.Fragment>
+                {matchingModules.length > 0 && (
+                    <div className={styles.SearchCategory}>Modules</div>
```

Comment:
Probably shouldn't have hover effects/hand cursor for these non-interactive "Modules" and "Streams" header things:

![image](https://user-images.githubusercontent.com/43438/54894076-e820e480-4ef3-11e9-89a7-744934ca1c93.png)

Alternatively these could perhaps be treated as accordions that default to open, so if I'm specifically looking for a stream I can just close the accordion?